### DISCUSSION TITLE
Tabularx

### DISCUSSION THREAD

**Discussion by alvesman** [joaomlourenco/novathesis#207] [Q&A] [answered]
Hi all and congratulations to the author. Amazing work.

I'm new to LaTex but overleaf has made it a very pleasant experience. My problem: I’m trying to build a multipage table using tabularx. I have it working fine on another LaTex document I have in overleaf but when I copy it to a chapter in the template it breaks down. The problem is when I add the code "endfirsthead", "endhead", to handle the breaks. I’m sure I’m missing something but I’m stuck…

If I remove the code delimited by %---------% it compiles. 

To reproduce the problem:
`
\captionof{table}{Caption of myTable}
\label{tab:myTable } 
\begin{tabularx}{1\textwidth}{
    >{\raggedright\arraybackslash}p{0.2\textwidth}
    >{\raggedright\arraybackslash}p{0.75\textwidth}
    }

%---------------------------------------------------------------%
\toprule
Field ID & Description \\
\midrule
\endfirsthead
%---------------------------------------------------------------%
\caption{Caption of myTable (continued)} \\
\toprule
Field ID & Description \\
\midrule
\endhead
%---------------------------------------------------------------%
\hline
\multicolumn{2}{r}{\footnotesize\textit{continued on the next page}}
\endfoot
% \bottomrule
\endlastfoot
%---------------------------------------------------------------%
    
\showrowcolors

Column 1, row 1 & Column 2, row 1 \\
Column 1, row 2 & Column 2, row 2 \\
Column 1, row 3 & Column 2, row 3 \\
Column 1, row 4 & Column 2, row 4 \\
Column 1, row 5 & Column 2, row 5 \\
Column 1, row 6 & Column 2, row 6 \\
Column 1, row 7 & Column 2, row 7 \\
Column 1, row 8 & Column 2, row 8 \\
Column 1, row 9 & Column 2, row 9 \\
Column 1, row 10 & Column 2, row 10 \\
Column 1, row 11 & Column 2, row 11 \\
Column 1, row 12 & Column 2, row 12 \\
Column 1, row 13 & Column 2, row 13 \\
Column 1, row 14 & Column 2, row 14 \\
Column 1, row 15 & Column 2, row 15 \\
Column 1, row 16 & Column 2, row 16 \\
Column 1, row 17 & Column 2, row 17 \\
Column 1, row 18 & Column 2, row 18 \\
Column 1, row 19 & Column 2, row 19 \\
Column 1, row 20 & Column 2, row 20 \\
Column 1, row 21 & Column 2, row 21 \\
Column 1, row 22 & Column 2, row 22 \\
Column 1, row 23 & Column 2, row 23 \\
Column 1, row 24 & Column 2, row 24 \\
Column 1, row 25 & Column 2, row 25 \\
Column 1, row 26 & Column 2, row 26 \\
Column 1, row 27 & Column 2, row 27 \\
Column 1, row 28 & Column 2, row 28 \\
Column 1, row 29 & Column 2, row 29 \\
Column 1, row 30 & Column 2, row 30 \\
Column 1, row 31 & Column 2, row 31 \\
Column 1, row 32 & Column 2, row 32 \\
Column 1, row 33 & Column 2, row 33 \\
Column 1, row 34 & Column 2, row 34 \\

\hiderowcolors
\hline
\end{tabularx}
`

Thanks in advance for any insights!
Manuel

**Comment by alvesman** [comment, marked as answer]
Solved by replacing **tabularx** with **xltabular**.

**Reply by b-pereira** [reply]
You've already found a solution! 

> I'm new to LaTex

For a newbie that is to say a lot!
 However, that maybe took some time to solve. Here is an alternative solution https://texblog.org/code-snippets/tables/#multipage, if it suits your needs (this is what I use for multipage tables).  In the same document, it specifies how to use other tables, like side by side tables, etc.
 
 Because you're a newbie, sometimes re-inventing the wheel can be a waste of time (or maybe not, if you're trying to learn more advanced stuff!). In here https://texblog.org/ you can find everything you want (this reference is already somewhere in this wiki and referenced in the README). By the way, there's a way to use CSV file contents directly in the same blog, also that I use too ... search for "table". What I mean directly is that you specify the CSV file in LaTeX, and then it's imported! Maybe you are already using it, but if you're using a script in this way it's a lot easier.
